src/qibo/quantum_info/metrics.py

```diff
@@ -852,6 +853,133 @@ def gate_error(channel, target=None, check_unitary: bool = False, backend=None):
     return error
 
 
+def diamond_norm(channel, target=None, **kwargs):
+    """Calculates the diamond norm :math:`\\|\\mathcal{E}\\|_{\\diamond}` of
+    ``channel`` :math:`\\mathcal{E}`. If a ``target`` channel :math:`\\Lambda`
+    is specified, then it calculates :math:`\\| \\mathcal{E} - \\Lambda\\|_{\\diamond}`.
+
+    Example:
+
+        .. testcode::
+
+            from qibo.quantum_info import diamond_norm, random_unitary, to_choi
+
+            nqubits = 1
+            dim = 2**nqubits
+
+            unitary = random_unitary(dim)
+            unitary = to_choi(unitary, order="row")
+
+            unitary_2 = random_unitary(dim)
+            unitary_2 = to_choi(unitary_2, order="row")
+
+            dnorm = diamond_norm(unitary, unitary_2)
+
+    Args:
+        channel (ndarray): row-vectorized Choi representation of a quantum channel.
+        target (ndarray, optional): row-vectorized Choi representation of a target
+            quantum channel. Defaults to ``None``.
+        kwargs: optional arguments to pass to CVXPY solver. For more information,
+            please visit CVXPY's API documentation
+            <https://www.cvxpy.org/api_reference/cvxpy.problems.html#problem>_.
+
+    Returns:
+        float: diamond norm of either ``channel`` or ``channel - target``.
+
+    .. note::
+        This function requires the optional CVXPY package to be installed.
+
+    """
+    try:  # pragma: no cover
+        import cvxpy  # pylint: disable=C0415
+    except ImportError:  # pragma: no cover
+        raise_error(
+            ModuleNotFoundError, "cvxpy module was not found. Please install it."
+        )
+
+    from scipy.sparse import eye as sp_eye  # pylint: disable=C0415
```

Comment:
What is the pylint issue here? 

By the way it is also fine to change this to something like
```py
from scipy import sparse
```
and move it as a global import, as it is good to avoid local imports as much as possible. Scipy is a dependency anyway so it won't fail.